rimage/lazy_encoded.go

```diff
@@ -99,28 +144,89 @@ func (lei *LazyEncodedImage) RawData() []byte {
 }
 
 // DecodedImage returns the decoded image.
-func (lei *LazyEncodedImage) DecodedImage() image.Image {
-	lei.decode()
-	return lei.decodedImage
+//
+// It is recommended to call DecodeImage and check for errors before using this method.
+func (lei *LazyEncodedImage) DecodedImage() (image.Image, error) {
+	err := lei.DecodeImage()
+	if err != nil {
+		return nil, err
+	}
+	return lei.decodedImage, nil
 }
 
 // ColorModel returns the Image's color model.
+//
+// It is recommended to call DecodeConfig and check for errors before using this method.
 func (lei *LazyEncodedImage) ColorModel() color.Model {
-	lei.decodeConfig()
+	err := lei.DecodeConfig()
+	if err != nil {
+		panic(err)
+	}
 	return lei.colorModel
 }
 
+// ColorModelSafe returns the Image's color model.
+//
+// This method is a safer alternative to the ColorModel method, as it provides error handling
+// instead of panicking, ensuring that the caller can manage any decoding issues gracefully.
+func (lei *LazyEncodedImage) ColorModelSafe() (color.Model, error) {
+	var model color.Model
+	err := safeCall(func() {
+		model = lei.ColorModel()
+	})
+	return model, err
+}
+
 // Bounds returns the domain for which At can return non-zero color.
 // The bounds do not necessarily contain the point (0, 0).
+//
+// It is recommended to call DecodeConfig and check for errors before using this method.
+// This method is considered unsafe as it will panic if the image is not decoded.
 func (lei *LazyEncodedImage) Bounds() image.Rectangle {
-	lei.decodeConfig()
+	err := lei.DecodeConfig()
+	if err != nil {
+		panic(err)
+	}
 	return *lei.bounds
 }
 
+// BoundsSafe returns the domain for which At can return non-zero color.
+// The bounds do not necessarily contain the point (0, 0).
+//
+// This method is a safer alternative to the Bounds method, as it provides error handling
+// instead of panicking, allowing the caller to handle any issues that arise during decoding.
+func (lei *LazyEncodedImage) BoundsSafe() (image.Rectangle, error) {
+	var bounds image.Rectangle
+	err := safeCall(func() {
+		bounds = lei.Bounds()
+	})
```

Comment:
I would recommend doiung this the exact opposite way.

Bounds() should be implemented in terms of BoundsSafe, no the other way around.

Same goes for all the other safe methods.

Reply:
oh ya that's nice